### PR TITLE
cfr improvements for javadoc, lambda param etc.

### PR DESCRIPTION
**Pull request by liach**
Fyi you can check at `build\sources\java` for the token replaced source with blossom. The git command works on my end and I effectively reverted 8c944b80a9dbd43c46f7b684e6d1154c2ba7a520

Replaces #2. @modmuss50 for review.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,10 +19,21 @@ archivesBaseName = 'cfr'
...
     }
 }
 
+java {
+    withSourcesJar()
+}
+
+blossom {
+    replaceTokenIn 'src-templates/org/benf/cfr/reader/util/CfrVersionInfo.java'
+    replaceToken '${project.version}', project.version
```
Comment (modmuss50) [inline build.gradle:36]: Ah nice, I should have looked into this. We should make it clear that this is a fork, possibly just append fabricmc/crf to the version metadata here?

Such as:

version + '+build.fabric-crf` or something idk.

**Review by modmuss50 — APPROVED** [review]
Im not familiar with this code base at all, so cannot really give it a decent review but nothing stands out to me.

**Comment by liach** [comment]
Now I copied the publishing stuff from mappingpoet. And seems this versioning works (currently `fabric-SNAPSHOT-0.0.2`)
image

**Comment by liach** [comment]
I think this pr is ready for squash merge as is.

File: build.gradle [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+    id "net.kyori.blossom" version "1.1.0"
 }
 
-sourceCompatibility = 1.6
-targetCompatibility = 1.6
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
 
-version = 'local'
+version = 'fabric-SNAPSHOT-0.0.2' // SNAPSHOT triggers commit hash info
```
Comment (modmuss50) [inline build.gradle]: this isnt really a valid version, id keep this as 0.0.2 and move the info else where tbh. IDK what you think?

Comment (liach) [reply build.gradle]: Sure, I can patch the template java file instead. Wait for me

**Comment by liach** [comment]
Done. How does this look now? 
image

**Comment by modmuss50** [comment]
Looks good, assuming `0.2.0+local` is the version on maven 👍

**Comment by liach** [comment]
local is added the same way in mappingpoet, i.e. build outside of github actions. If you decide to merge and build from github manual workflow I suggest you upload secrets here (like in mappingpoet)

File: build.gradle [hunk 3]
```diff
@@ -1,27 +1,37 @@
 plugins {
     id 'java'
     id 'maven-publish'
+    id "net.kyori.blossom" version "1.1.0"
 }
 
-sourceCompatibility = 1.6
-targetCompatibility = 1.6
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
```
Comment (modmuss50) [inline build.gradle:8]: This needs to also set -release 8 (see another project) as its building on never java.

Comment (liach) [reply build.gradle:8]: Ah, so like
```gradle
if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_1_8) {
    tasks.withType(JavaCompile) {
        options.release = 8
    }
}
```
?

Comment (modmuss50) [reply build.gradle:8]:
```gradle
tasks.withType(JavaCompile).configureEach {
	if (JavaVersion.current().isJava9Compatible()) {
		 it.options.release = 8
	}
}
```

Is slighly nicer, but yes.

**Comment by modmuss50** [comment]
Looks good now. 👍

**Comment by modmuss50** [comment]
https://maven.fabricmc.net/net/fabricmc/cfr/0.0.2/

**Pull request merged**